### DISCUSSION TITLE
Can RDP protocol graphics be converted to bitmap or video ？

### DISCUSSION THREAD

**Discussion by Mr-LZG** [neutrinolabs/xrdp#2664] [Q&A]
Can RDP protocol graphics be converted to bitmap or video ？ I want to record RDP remote connection and play it back。
 I currently have a solution that records the raw data of the RDP protocol and can play it back through the RDP client. However, the RDP original protocol data is too large, and it cannot be dragged freely like a video.

**Comment by matt335672** [comment]
There is no standard format for RDP recording that I'm aware of, so I don't think you'll find a solution to your problem as you describe it.

You're not describing your actual problem, but it sounds like you're exploring possible solutions to a problem. It's possible there may be a commercial product which could be adapted to suit your use-case. This smells like an enterprise application, as you'll need to MITM the connection to get access to the RDP datastream.

Another option is to present your desktop over VNC and convert that into a video using vnc2flv. You wouldn't get sound that way.

Sorry I can't be any more help.

**Reply by Mr-LZG** [reply]
I use XRDP as a proxy, and there are many people accessing rdp or vnc resources at the same time. I want to record all the operations of these people (video only) for post-event auditing.

**Reply by Mr-LZG** [reply]
Is it possible for me to convert RDP protocol data into video data, and if possible, which protocols should I pay attention to.

**Reply by matt335672** [reply]
I'm not aware of any open source product which would let you do this I'm afraid.

**Reply by fuqifacai** [reply]
> Is it possible for me to convert RDP protocol data into video data, and if possible, which protocols should I pay attention to.

Anyway ,you could capture screen shot of RDP. save as 24fps .should be a video.
